### PR TITLE
[Typing][A-74] Add type annotations for `paddle/vision/models/densenet.py`

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### PR Category

User Experience
### PR Types

Improvements

### Description

Add type annotations for `paddle/vision/models/densenet.py`
- #65008

### CODE REVIEW

File: python/paddle/vision/models/densenet.py [hunk 1]
```diff
@@ -334,7 +368,12 @@ def forward(self, input):
         return y
 
 
-def _densenet(arch, layers, pretrained, **kwargs):
+def _densenet(
+    arch: _DenseNetArch,
+    layers: int,
+    pretrained: bool,
+    **kwargs: _DenseNetKwargs,
```
Comment (megemini) [inline python/paddle/vision/models/densenet.py]: `Unpack[_DenseNetKwargs]`

后面几个也一样 ～

**Review by megemini — COMMENTED** [review]
需要 `from __futrue__ import annotations`

File: python/paddle/vision/models/densenet.py [hunk 2]
```diff
@@ -29,9 +31,29 @@
...
+        "densenet169",
+        "densenet201",
+        "densenet264",
+    ]
+
+    class _DenseNetKwargs(TypedDict):
+        bn_size: int
+        dropout: float
+        num_classes: int
+        with_pool: bool
```
Comment (megemini) [inline python/paddle/vision/models/densenet.py]: 这几个参数是不是也改为 `NotRequired` ？

Comment (Asthestarsfalll) [reply python/paddle/vision/models/densenet.py]: 已修改，刚才没有push上来

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**